src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -88,10 +89,17 @@ internal sealed class PeWriter
             // If PDB writer is given, we have to have PDB path.
             Debug.Assert(nativePdbWriterOpt == null || pdbPathOpt != null);
 
-            var peWriter = new PeWriter(context.Module.Properties, context.Module.Win32Resources, context.Module.Win32ResourceSection, pdbPathOpt, deterministic);
-            var mdWriter = FullMetadataWriter.Create(context, messageProvider, allowMissingMethodBodies, deterministic, getPortablePdbStreamOpt != null, cancellationToken);
+            try
+            {
+                var peWriter = new PeWriter(context.Module.Properties, context.Module.Win32Resources, context.Module.Win32ResourceSection, pdbPathOpt, deterministic);
+                var mdWriter = FullMetadataWriter.Create(context, messageProvider, allowMissingMethodBodies, deterministic, getPortablePdbStreamOpt != null, cancellationToken);
 
-            return peWriter.WritePeToStream(mdWriter, getPeStream, getPortablePdbStreamOpt, nativePdbWriterOpt);
+                return peWriter.WritePeToStream(mdWriter, getPeStream, getPortablePdbStreamOpt, nativePdbWriterOpt);
+            }
+            catch (Exception ex) when (!(ex is PdbWritingException || ex is ResourceException || ex is PermissionSetFileReadException))
```

Comment:
This code path already blanket catches exceptions in many areas, including the act of opening this very stream.  This change is normalizing the behavior, not adding a new case.  It makes the behavior consistent for all IO errors during Emit.  

> So are you officially signing up for investigating crashes in PE, metadata and PDB writers without a full stack in the crash dump? ;)

Ty is closing the hole on the number one Watson in the Roslyn code base.  It's so prolific I've had to dupe at least four Watson bugs against this coed path.  He's actively responding to existing crashes.  If new ones pop up and are not actionable based on just the stack then we can look to further instrumenting those code paths.